### PR TITLE
Invoice Screen: UI Enhancement and PDF Upload Feature

### PR DESCRIPTION
**Pull request by TejasDhole**
Invoice Screen:
1.check invoice ui
2.can see muliple img wil miniature view
3.reduce code reduncy 
4.updated color to used from color file
5.pull to reefresh feature added 
3.added refresh on clicking invocie screen 
7.added skelaton view for the table (shimmer)

Add Invocie Screen 
1.upgraded uplaod bill ui
2.fixed upload invoice bill in pdf format and convert into img byte and post api
3.fixed the insuuee to send multiple pages from pdf
4 added skelaton view in table (shiimmer)

### CODE REVIEW

**Comment by Anik-Chakraborty** [comment]
image
Browse file is not opening any window to pick files.

File: lib/functions/documentApi/postDocumentApiCall.dart [hunk 1]
```diff
@@ -15,15 +15,15 @@ postDocumentApiCall(Map datanew2) async {
     //     {"documentType": documentType, "data": photo64code}
```
Comment (Anik-Chakraborty) [inline lib/functions/documentApi/postDocumentApiCall.dart:15]: Remove such commented codes.

File: lib/functions/documentApi/postDocumentApiCall.dart [hunk 2]
```diff
@@ -15,15 +15,15 @@ postDocumentApiCall(Map datanew2) async {
...
+    // print(datanew2);
     String body = json.encode(datanew2);
     final String documentApiUrl = dotenv.get('documentApiUrl').toString();
     final response = await http.post(Uri.parse("$documentApiUrl"),
         headers: <String, String>{
           'Content-Type': 'application/json; charset=UTF-8',
         },
         body: body);
-    print(response.body);
+    // print(response.body);
```
Comment (Anik-Chakraborty) [inline lib/functions/documentApi/postDocumentApiCall.dart:26]: Here also

File: lib/functions/invoiceApi/invoiceApiService.dart [hunk 3]
```diff
@@ -15,14 +15,15 @@ class InvoiceApiService {
       return data;
     } else {
       debugPrint('Error Status Code: ${response.statusCode}');
-      debugPrint('Error Response Body: ${response.body}');
+      // debugPrint('Error Response Body: ${response.body}');
```
Comment (Anik-Chakraborty) [inline lib/functions/invoiceApi/invoiceApiService.dart:18]: ..

File: lib/functions/invoiceApi/invoiceApiService.dart [hunk 4]
```diff
@@ -53,7 +55,7 @@ class InvoiceApiService {
 
       if (responseData.containsKey('invoiceId')) {
         String invoiceId = responseData['invoiceId'].toString();
-        print('Invoice created successfully! Invoice ID: $invoiceId');
+        // print('Invoice created successfully! Invoice ID: $invoiceId');
```
Comment (Anik-Chakraborty) [inline lib/functions/invoiceApi/invoiceApiService.dart:58]: ..

File: lib/screens/invoiceScreens/InvoiceScreen.dart [hunk 5]
```diff
@@ -507,11 +575,36 @@ class _InvoiceScreenState extends State<InvoiceScreen> {
...
+    return Shimmer.fromColors(
+      baseColor: Colors.grey[300]!,
+      highlightColor: Colors.grey[100]!,
+      child: Container(
+        height: 500, // Adjust the height as needed
+        margin: EdgeInsets.all(16.0),
+        child: Column(
+          children: List.generate(
+            6, // Number of shimmer items you want
+            (index) => Container(
```
Comment (Anik-Chakraborty) [inline lib/screens/invoiceScreens/InvoiceScreen.dart:597]: The same widget is present on add_invoice_screen.dart file... so reuse it...and define it in a  widget directory

**Review by Anik-Chakraborty — CHANGES_REQUESTED** [review]

File: lib/functions/documentApi/postDocumentApiCall.dart (hunk 1 above)
Comment (TejasDhole) [reply lib/functions/documentApi/postDocumentApiCall.dart:15]: This code was added by another developer, and its purpose isn't clear to me.
 I considered removing it during development, but I wasn't aware of its potential use in the future
 Please review and decide whether to keep, modify, or remove this code based on its original purpose.

**Review by Anik-Chakraborty — APPROVED** [review]

**Pull request merged**